src/core/net/srp_client.cpp

```diff
@@ -901,6 +901,8 @@ Error Client::ReadOrGenerateKey(Crypto::Ecdsa::P256::KeyPair &aKeyPair)
         }
     }
 
+    // Initialise the keypair before generating.
+    aKeyPair.Init();
     SuccessOrExit(error = aKeyPair.Generate());
```

Comment:
How about doing the `Init()` call  in `Generate()` instead?

Reply:
I thought of it but decided against it as it would looks messy to call Init from multiple places inside the kayPair class. If that is OK with OT design I am OK with doing that.

Reply:
It is perfectly ok to call `Init()` from inside `KeyPair` (actually we can cosnider making it `private` method?)

Reply:
Changed in the latest push.